openfl/federated/data/sources/pytorch/s3_folder.py

```diff
@@ -0,0 +1,71 @@
+# Copyright 2020-2025 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+import os
+from urllib.parse import urlparse
+
+import boto3
+
+from openfl.federated.data.sources.pytorch.folder_dataset import FolderDataset, LabelMapper
+
+
+class S3Folder(FolderDataset):
+    def __init__(
+        self,
+        uri,
+        label_mapper: LabelMapper,
+        endpoint=None,
+        access_key_env_name=None,
+        secret_key_env_name=None,
+        transform=None,
+    ):
+        """
+        Args:
+            uri (str): URI to the S3 object.
+            label_mapper (LabelMapper): LabelMapper object to map class names to indices.
+            transform (callable, optional): Transformations to apply to images.
+        """
+        self.uri = uri
+        self.endpoint = endpoint
+        self.access_key_env_name = access_key_env_name
+        self.secret_key_env_name = secret_key_env_name
+        super().__init__(label_mapper, transform=transform)
+
+    def _load_samples(self):
+        """Loads all file paths and their inferred labels into self.samples."""
+        parsed = urlparse(self.uri)
+        bucket_name = parsed.netloc
+        prefix = parsed.path.lstrip("/")
+        access_key_env_name = (
+            os.environ.get(self.access_key_env_name) if self.access_key_env_name else None
+        )
+        secret_key_env_name = (
+            os.environ.get(self.secret_key_env_name) if self.secret_key_env_name else None
+        )
```

Comment:
this code section is a bit confusing - are `access_ken_env_name` and `secret_key_env_name` names of env variables, or their content?